lib/test/unit/assertions.rb

```diff
@@ -934,7 +934,12 @@ def assert_in_delta(expected_float, actual_float, delta=0.001, message="")
                                                   delta,
                                                   message)
           assert_block(full_message) do
-            (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
+            abs_diff = (expected_float - actual_float).abs rescue nil
+            if abs_diff && abs_diff.respond_to?(:<=)
+              abs_diff <= delta
+            else
+              (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
+            end
```

Comment:
@kou How do you think this way?

Reply:
Another candidate:

```ruby
abs_diff = (expected_float - actual_float).abs rescue nil
begin
  abs_diff <= delta
rescue
  begin
    !(delta > abs_diff)
  rescue
    (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
  end
end
```

Reply:
How about this?

```ruby
begin
  (expected_float - actual_float).abs <= delta
rescue
  (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
end
```

Reply:
It's very simple but doesn't depend on `delta`'s methods.  Yes, I like this way.